engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamPlatform.java

```diff
@@ -152,6 +154,14 @@ protected void onActorStarted() {
       final var startRecoveryTimer = metrics.startRecoveryTimer();
       final long snapshotPosition = recoverFromSnapshot();
 
+      final TransactionContext transactionContext = zeebeDb.createContext();
+
+      final var partitionId = processingContext.getPartitionId();
+      final var zeebeState = new ZeebeDbState(partitionId, zeebeDb, transactionContext);
```

Comment:
This is for me scope of the engine, which I moved it before to the engine init method.